sympy/functions/special/zeta_functions.py

```diff
@@ -384,7 +385,11 @@ def _eval_nseries(self, x, n, logx, cdir=0):
                     s.append(term/k**nu)
                 return Add(*s) + o
 
-        return super(polylog, self)._eval_nseries(x, n, logx, cdir)
+        result = super(polylog, self.expand())._eval_nseries(x, n, logx, cdir)
```

Comment:
It seems like this expansion is something that should be more targetted than just applying the general expansion -- what is self and what do you want expand to do?

Reply:
+1
Even I wasn't sure of the change

Reply:
This change isn't related to the issue directly though,what I noticed while testing my code for different polynomial expressions(z) like `(x-1)**2` which are directly not taken care of by the polylog _eval_nseries and instead call to the `super()` is made the gamma term is added twice to the result, the reason being that when the call to the super is made twice due to the following condition https://github.com/sympy/sympy/blob/580478123816a986b27d27f379ad3d0482af2c82/sympy/core/function.py#L722C1-L750C49
after the expression is expanded it again calls the polylog _eval_nseries which again calls super which then computes the series and in the process the gamma term is added twice.

Reply:
What my idea is that since when super() is called and the argument that is being passed is finite around zero most probably there maybe no harm in expanding the expression before hand as it will be done nonetheless in super as observed,moreover since the polylog expansion can be expressed in closed form as shown [here](https://en.wikipedia.org/wiki/Polylogarithm#Series_representations) we can simply add the missing gamma term to complete the expansion.

Reply:
can we do without the `expand()` here

Reply:
Is the call to expand here needed or not? Does this code otherwise not work without it?